### PR TITLE
openapi Schema derive required

### LINKED ISSUE #50: Information on required fields is missing in generated structures
I create simple handler:

```
#[derive(Debug, Default, Serialize, Schema)]
struct Person0 {
    name: String,
    age: u64,
    city: Option<String>,
}

#[derive(Debug, Default, Serialize, Schema)]
struct Person {
    name: String,
    age: u64,
    city: Option<String>,
    per0: Person0,
}

#[rweb::get("/test")]
async fn aaa(#[data] appAddr: Arc<App>) -> Result<rweb::Json<Person>, Infallible> {

    Ok(Person {
        name: "Baaaa".into(),
        age: 444,
        //city: Some(String::from("dasdas"))
        city: None,
        per0: Person0 {
            name: "dsadas".into(),
            age: 44,
            city: Some(String::from("dsdas")),
        }
    }.into())
}
```

The following specification was generated from this handler:

```
openapi: 3.0.1
info:
  title: ''
  version: ''
paths:
  /test:
    get:
      responses:
        '200':
          description: ''
          content:
            application/json:
              schema:
                type: object
                properties:
                  name:
                    type: string
                  age:
                    type: integer
                  city:
                    type: string
                    nullable: true
                  per0:
                    type: object
                    properties:
                      name:
                        type: string
                      age:
                        type: integer
                      city:
                        type: string
                        nullable: true
```

Information that all fields are required is missing from the generated specification.

The correct specification should look like this:

```
openapi: 3.0.1
info:
  title: ''
  version: ''
paths:
  /test:
    get:
      responses:
        '200':
          description: ''
          content:
            application/json:
              schema:
                type: object
                required:
                  - name
                  - age
                  - per0
                properties:
                  name:
                    type: string
                  age:
                    type: integer
                  city:
                    type: string
                    nullable: true
                  per0:
                    type: object
                    required:
                    - name
                    - age
                    - city
                    properties:
                      name:
                        type: string
                      age:
                        type: integer
                      city:
                        type: string
                        nullable: true
```

Can I ask to fix this error ?
I have analysed the "Scheme" macro, but am not yet able to make a fix for this error.

### PR DESCRIPTION
**Pull request by E-gy**
TLDR: Fixes #50 

For both enums and structs, populates derived schema's `required` property with all non-nullable fields.
in other words, all fields except those that are nullable.

The behaviour can be further improved by also ignoring fields that have serde `skip_serializing_if` attribute.
Though technically speaking it would be correct to _only_ exclude fields that have `skip_serializing_if` attribute (which will allow the spec to contain nullable required fields).

### CODE REVIEW

**Comment by E-gy** [comment]
For reference:
- [OpenAPITools' client generator for Rust](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/rust/model.mustache#L66) treats both required-nullable and not-required fields in the same way - by wrapping them into `Option`. Here, a server with
```rs
#[derive(Schema)]
struct Stru {
  #[serde(skip_serializing_if = "Vec::is_empty")]
  pub vec: Vec<String>
}
```
(if we use `skip_serializing_if` attribute for `required` fields) will produce Rust client
```rs
#[derive(Schema)]
struct Stru {
  pub vec: Option<Vec<String>>
}
```
- The required-but-nullable is mostly important TS/JS family of languages

File: macros/src/openapi/derive.rs [hunk 1]
```diff
@@ -221,6 +221,45 @@ fn handle_fields(type_attrs: &[Attribute], fields: &mut Fields) -> Block {
...
+                Default::default(),
+            )
+        })
+        .collect();
+
+    if reqf_v.is_empty() {
+        q!({ vec![] })
+    } else {
+        q!(Vars { reqf_v }, {
+            vec![reqf_v].into_iter().flatten().collect()
```
Comment (kdy1) [inline macros/src/openapi/derive.rs:257]: Doesn't `vec![reqf_v]` work?

**Review by kdy1 — CHANGES_REQUESTED** [review]
Mostly LGTM, but one issue.

File: macros/src/openapi/derive.rs (hunk 1 above)
Comment (E-gy) [reply macros/src/openapi/derive.rs:257]: Here `reqf_v` are elements typed `Option<Str>` (see [l240-242](https://github.com/kdy1/rweb/pull/59/files#diff-652ac060d5f66badbfcf5b3305d2cfd8b38157138a92f93e4b0f38090da1a135R240-R242)). So we have `Vec<Option<Str>>` and want `Vec<Str>`.
